### ISSUE TITLE
MongooseArray.pull causes query to include where versioning.

### ISSUE THREAD

**Issue by dan-perron** [Automattic/mongoose#6190]
We have a model of a user like
```
const User = mongoose.model(
    'User', 
    new mongoose.Schema({
        unreadPosts: [{type: mongoose.Schema.Types.ObjectId}]
    })
);
```
When a user accesses a post we execute some code like:
```
user.unreadPosts.pull(postId);
user.save();
```

This causes an update like:
```
	"op" : "update",
	"ns" : "foo.users",
	"query" : {
		"_id" : ObjectId("58fe0d8e262b253b29a7e020"),
		"__v" : 831
	},
	"updateobj" : {
		"$pullAll" : {
			"postIds" : [
				ObjectId("5977b8ccc22a2404fa8b9908")
			]
		},
		"$set" : {
			"updated" : ISODate("2018-03-01T19:26:59.574Z")
		},
		"$inc" : {
			"__v" : 1
		}
	},
```

I expect this to cause the versionKey to be incremented (it could change array positions) but I don't think it needs the versionKey in the query.

node v8.9.1
mongoose 5.0.3
MongoDB 3.4

**Comment by vkarpov15** [comment]
It's worth discussing but I can see the rationale behind keeping `__v` in the query in this case. There's nothing preventing you from having multiple objects in an array with the same `_id`, so if the array changed underneath then `$pullAll` could be pulling an unexpected document. I'd love to hear arguments to the contrary though.

**Comment by dan-perron** [comment]
I guess I'm not sure why someone would be using `.pull()` if they didn't expect all copies with that `_id` removed. Are you suggesting something like 
```
if (objectsWithIdCount > 1) {
  // remove one of the documents by index
} else {
  document.array.pull(id);
}
```

**Comment by vkarpov15** [comment]
Yeah that's a fair point. I think the only reason why things are this way right now is because mongoose consistently sets the version key in the query if you remove subdocs from the array, but in this case it shouldn't be necessary. Will investigate for a future release :+1:

**Comment by vkarpov15** [comment]
I investigated and it looks like this is safe. Fix will be in 5.0.15 :+1:
